### ISSUE TITLE
remember() not working

### ISSUE THREAD

**Issue by TrulyBright** [aio-libs/aiohttp-security#364]
I inserted some print()s to the api:

```
async def remember(self, request, response, identity, **kwargs):
        session = await get_session(request)
        session[self._session_key] = identity
        print(dict(session))
```
```
async def identify(self, request):
        session = await get_session(request)
        print(dict(session))
        return session.get(self._session_key)
```
Below are respective result:
```
{'AIOHTTP_SECURITY': 'username'}
```
```
{}
```
which mean the `AIOHTTP_SECURITY` session is gone before `identify()` is called. This seem to be a bug.

**Comment by GenyaSol** [comment]
> I inserted some print()s to the api:
> 
> ```
> async def remember(self, request, response, identity, **kwargs):
>         session = await get_session(request)
>         session[self._session_key] = identity
>         print(dict(session))
> ```
> 
> ```
> async def identify(self, request):
>         session = await get_session(request)
>         print(dict(session))
>         return session.get(self._session_key)
> ```
> 
> Below are respective result:
> 
> ```
> {'AIOHTTP_SECURITY': 'username'}
> ```
> 
> ```
> {}
> ```
> 
> which mean the `AIOHTTP_SECURITY` session is gone before `identify()` is called. This seem to be a bug.

Show your login handler. set_cookie method doesn't work when response raised. You should use return instead.

**Comment by Dreamsorcerer** [comment]
> Show your login handler. Method set_cookie doesn't work when response raised. You should use return instead.

This is a known regression in 3.7 with HTTPException, it is fixed in 3.8. Other than to work around this regression, you should always raise HTTPExceptions, returning them is deprecated.

**Comment by GenyaSol** [comment]
> > Show your login handler. Method set_cookie doesn't work when response raised. You should use return instead.
> 
> This is a known regression in 3.7 with HTTPException, it is fixed in 3.8. Other than to work around this regression, you should always raise HTTPExceptions, returning them is deprecated.

I don't see 3.8 aiohttp release on PYPI. If you mean Python 3.8 then it works the same.

**Comment by derlih** [comment]
There is an issue in `aiohttp` for it.
https://github.com/aio-libs/aiohttp/issues/5366
